numpy/_core/arrayprint.py

```diff
@@ -1468,10 +1454,11 @@ def _void_scalar_to_string(x, is_repr=True):
     scalartypes.c.src code, and is placed here because it uses the elementwise
     formatters defined above.
     """
-    options = _format_options.copy()
+    options = _format_options.get().copy()
 
     if options["legacy"] <= 125:
-        return StructuredVoidFormat.from_data(array(x), **_format_options)(x)
+        return StructuredVoidFormat.from_data(
+            array(x), **_format_options.get())(x)
```

Comment:
Btw. this is what I meant with: maybe we should change it and just not pass anything to the formatters.  It seemed messy last time I looked at this code, and it *is* global state pretty much.  Maybe it easier to accept that (and in the few places where we mutate it, just mutate the contextvar).

But, looking through, since the contextvar is just one dict/object (and it is not clear to me that stopping that would do any good), there seems little use in doing it here now, unless you want to make the code better.